WordPress/gutenberg
# Add root components directory

Previously: #716

This pull request expands upon the restructuring outlined in #716, creating a new common root `components` directory containing components which have shown to be generic between blocks- and editor-specific components.

__Testing instructions:__

Ensure the build completes, tests pass, and editor initializes and appears without regressions. Specifically observe that there are no regressions in displayed buttons, icon buttons, icons, and toolbar components.
